proto/info/info.proto

```diff
@@ -214,6 +214,10 @@ enum ZSwState {
   // START_DELAYED state after ERROR.
   // Had to prefix with Z_SW_STATE_ to satisfy yetus checks.
   START_DELAYED = 22; // App has been configured to start with delay by user
+
+  PENDING = 23; // Kubernetes app in pending state
+  APPFAILED  = 24; // App failed to start
```

Comment:
The fact that FAILED exists and is used in some object for BaseOS doesn't mean it can't be used in other objects such as the app instance. So I don't think you need to add a uniq APPFAILED value here.

Reply:
But the make proto compilation failed. That's why I had to change. 

bash-3.2$ make proto
rm -rf go/*/; mkdir -p go # building proto-api-go
protoc -I./proto --go_out=paths=source_relative:./go proto/*/*.proto
info/info.proto:860:3: "FAILED" is already defined in "org.lfedge.eve.info".
info/info.proto:860:3: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "FAILED" must be unique within "org.lfedge.eve.info", not just within "BaseOsStatus".

Reply:
Ah - I missed that it was in another enum definition.

Do you really need a *FAILED or can you use the existing ERROR enum value?

Reply:
FAILED is returned by kubernetes as one the states for the app. We better be explicit and return same error. I looked at other options and only thing that comes close is HALTED, which does not convey the failed state.

Reply:
Why can't you use the existing ERROR?
When used on an app instance there is the associated
repeated ErrorInfo appErr = 14;
which enables you to specify additional details as to the nature/reason for the error.

Reply:
IF you have a sample PR I look into that ?

Reply:
I looked at the code deeply.   repeated ErrorInfo appErr = 14; is the detailed error description for an error state.
 I am creating new error states, so not clear to me on how to proceed here.

Reply:
> I looked at the code deeply. repeated ErrorInfo appErr = 14; is the detailed error description for an error state. I am creating new error states, so not clear to me on how to proceed here.

We map the pillar strate to the API state in https://github.com/lf-edge/eve/blob/d72bdf814fc59d8519ccb0c9fec2981664c310e9/pkg/pillar/types/types.go#L111

(You need to extend that function in any case to avoid the log.Fatal).

There you can map FAILED to ERROR.
If you want to set the error info you need to do that in the caller in https://github.com/lf-edge/eve/blob/d72bdf814fc59d8519ccb0c9fec2981664c310e9/pkg/pillar/cmd/zedagent/handlemetrics.go#L1061

Reply:
Yes I extended the types.go function to handle fatal errors. Ok then I will removed APPFAILED from this PR and map FAILED to ERROR in types.go.  That should address your questions I guess.

Reply:
Force updated the commits